src/PhysicsEngine.cpp

```diff
@@ -185,3 +184,257 @@ void ForwardEuler::march_step(double tStamp, double tStep) {
     rocket_.set_t_net(t_net_if);
     rocket_.set_q_ornt(q_ornt);
 }
+
+/**
+ * @brief Averages forces and moments for a more accurate euler step
+ * 
+ * @param tStamp 
+ * @param tStep 
+ */
+void RungeKutta::march_step(double tStamp, double tStep) {
+
+    /*************** Retrieve Instantaneous Rocket Parameters *****************/
+
+    Vector3 pos_if = rocket_.get_r_vect();
+    Vector3 vel_if = rocket_.get_r_dot();
+    Vector3 accel_if = rocket_.get_r_ddot();
+    Vector3 ang_vel_if = rocket_.get_w_vect();
+    Vector3 ang_accel_if = rocket_.get_w_dot();
+    
+    Quaternion<double> orient = rocket_.get_q_ornt();
+
+    double inertia[9];         // moments of inertia
+    rocket_.get_I(inertia);
+
+    double mass = rocket_.get_mass();
+
+    /********************* Calculate Intermediate States ***********************/
+    // Each state is used to calculate the next state
+
+    State k1 {vel_if, accel_if, ang_vel_if, ang_accel_if};           // Reformating of the initial state of the rocket
```

Comment:
k1 needs to be recalculated with the calc_state function here. vel_if and friends are calculated before all the derivatives from the previous iteration are applied so the derivative needs to be recomputed.

Reply:
You can't call calc_state for k1.  One of the parameters for calc_state is a State, since the function bases its calculations on the initial values pulled from the rocket class and the information stored in the State class.  Since k1 is the state the rocket is starting in, this function can't be called because no other States exist to pass as a parameter.  It also does not need to be recalculated since k1 is just the state that the rocket is currently in, which was calculated by the last RungeKutta step and doing that math again would be redundant.